addons/mail/models/mail_template.py

```diff
@@ -466,8 +471,12 @@ def send_mail(self, res_id, force_send=False, raise_exception=False, email_value
         :param str notif_layout: optional notification layout to encapsulate the
             generated email;
         :returns: id of the mail.mail that was created """
+
+        # Grant access to send_mail only if access to related document
         self.ensure_one()
-        Mail = self.env['mail.mail']
+        self._send_check_access([res_id])
+
+        Mail = self.env['mail.mail'].sudo()
```

Comment:
Looks like it's only used once.
Maybe directly call `self.env['mail.mail'].sudo()` below instead, and rename variable to `mail_sudo` for clarity?

Reply:
Done for the variable removal part; mail is still mail because all mails will be sudo now, so, hey, bisous.